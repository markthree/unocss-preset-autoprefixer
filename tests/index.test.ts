import { createGenerator, escapeSelector } from 'unocss'
import presetUno from '@unocss/preset-uno'
import { describe, expect, test } from 'vitest'
import presetPrefixer from '../src'

describe('preset-prefixer', () => {
  const uno = createGenerator({
    presets: [presetUno(), presetPrefixer()],
  })

  test('flex autoprefixer', async () => {
    const { css } = await uno.generate(['flex'], { preflights: false })

    expect(css).toMatchSnapshot()
  })

  test('text space', async () => {
    const { css } = await uno.generate(['text-1'], { preflights: false })

    expect(css).toMatchSnapshot()
  })

  test('font weight', async () => {
    const { css } = await uno.generate(['font-400', 'font-bold'], { preflights: false })

    expect(css).toMatchSnapshot()
  })

  test('background gradient', async () => {
    const { css } = await uno.generate(['bg-gradient-to-tr', 'from-purple-500', 'to-pink-500'], { preflights: false })

    expect(css).toMatchSnapshot()
  })

  test('p-x', async () => {
    const { css } = await uno.generate(['px-10'], { preflights: false })

    expect(css).toMatchSnapshot()
  })
  test('space-x', async () => {
    const { css } = await uno.generate(['space-x-10'], { preflights: false })

    expect(css).toMatchSnapshot()
  })
})
