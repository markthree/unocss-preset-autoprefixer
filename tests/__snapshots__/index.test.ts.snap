// Vitest Snapshot v1

exports[`preset-prefixer > background gradient 1`] = `
"/* layer: default */
.from-purple-500{--un-gradient-from:rgba(168,85,247,var(--un-from-opacity, 1));--un-gradient-to:rgba(168,85,247,0);--un-gradient-stops:var(--un-gradient-from), var(--un-gradient-to);}
.to-pink-500{--un-gradient-to:rgba(236,72,153,var(--un-to-opacity, 1));}
.bg-gradient-to-tr{--un-gradient-shape:to top right;--un-gradient:var(--un-gradient-shape), var(--un-gradient-stops);background-image:linear-gradient(var(--un-gradient));}"
`;

exports[`preset-prefixer > flex autoprefixer 1`] = `
"/* layer: default */
.flex{display:-webkit-box;display:-webkit-flex;display:flex;}"
`;

exports[`preset-prefixer > font weight 1`] = `
"/* layer: default */
.font-400{font-weight:400;}
.font-bold{font-weight:700;}"
`;

exports[`preset-prefixer > p-x 1`] = `
"/* layer: default */
.px-10{padding-left:2.5rem;padding-right:2.5rem;}"
`;

exports[`preset-prefixer > px 1`] = `
"/* layer: default */
.px-10{padding-left:2.5rem;padding-right:2.5rem;}"
`;

exports[`preset-prefixer > space-x 1`] = `
"/* layer: default */
.space-x-10>:not([hidden])~:not([hidden]){--un-space-x-reverse:0;margin-left:calc(2.5rem*calc(1 - var(--un-space-x-reverse)));margin-right:calc(2.5rem*var(--un-space-x-reverse));}"
`;

exports[`preset-prefixer > text space 1`] = `
"/* layer: default */
.text-1{font-size:.25rem;}"
`;
